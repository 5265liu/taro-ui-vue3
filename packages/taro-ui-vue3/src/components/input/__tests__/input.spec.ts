import { mount } from '@vue/test-utils'
import AtInput from '../index.vue'

const factory = (values = {}, slots = { default: [] }) => {
  return mount(AtInput, {
    slots,
    props: { ...values },
  })
}

describe('AtInput Snap', () => {
  it('render initial AtInput', () => {
    const wrapper = factory()
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props style', () => {
    const wrapper = factory({ style: 'border:none;' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props value', () => {
    const wrapper = factory({ value: 'value' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props name', () => {
    const wrapper = factory({ name: 'name' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props placeholder', () => {
    const wrapper = factory({ placeholder: 'placeholder' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props title', () => {
    const wrapper = factory({ title: 'title' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props maxLength', () => {
    const wrapper = factory({ maxLength: 10 })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props type(number)', () => {
    const wrapper = factory({ type: 'number' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props type(password)', () => {
    const wrapper = factory({ type: 'password' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props type(idcard)', () => {
    const wrapper = factory({ type: 'idcard' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props type(digit)', () => {
    const wrapper = factory({ type: 'digit' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props type(phone)', () => {
    const wrapper = factory({ type: 'phone' })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props disabled', () => {
    const wrapper = factory({ disabled: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props border', () => {
    const wrapper = factory({ border: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props editable', () => {
    const wrapper = factory({ editable: false })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props error', () => {
    const wrapper = factory({ error: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props clear', () => {
    const wrapper = factory({ clear: true })
    expect(wrapper.element).toMatchSnapshot()
  })

  it('render AtInput -- props required', () => {
    const wrapper = factory({ required: true })
    expect(wrapper.element).toMatchSnapshot()
  })
})

describe('AtInput Event', () => {
  it('AtInput onChange', () => {
    const onChange = jest.fn()
    const wrapper = factory({ onChange: onChange })
    wrapper.find('.at-input__input').trigger('input', { detail: { value: 'value' } })
    expect(onChange).toBeCalled()
  })

  it('AtInput onFocus', () => {
    const onFocus = jest.fn()
    const wrapper = factory({ onFocus: onFocus })
    wrapper.find('.at-input__input').trigger('focus', { height: 30 })
    expect(onFocus).toBeCalled()
  })

  it('AtInput onConfirm', () => {
    const onConfirm = jest.fn()
    const wrapper = factory({ onConfirm: onConfirm })
    wrapper.find('.at-input__input').trigger('confirm', { detail: { value: 'value' } })
    expect(onConfirm).toBeCalled()
  })

  it('AtInput onClick', () => {
    const onClick = jest.fn()
    const wrapper = factory({ onClick: onClick, editable: false })
    wrapper.find('.at-input .at-input__overlay').trigger('tap')
    expect(onClick).toBeCalled()
  })

  it('AtInput clear Behavior', () => {
    const onChange = jest.fn()
    const wrapper = factory({ onChange: onChange, clear: true, value: 'value' })
    wrapper.find('.at-input .at-input__icon').trigger('touchEnd')
    expect(wrapper.vm.inputClearing).toBeTruthy()
    expect(onChange).toBeCalled()
  })

  it('AtInput error Behavior', () => {
    const onErrorClick = jest.fn()
    const wrapper = factory({ onErrorClick: onErrorClick, error: true, value: 'value' })
    wrapper.find('.at-input .at-input__icon').trigger('touchStart')
    expect(onErrorClick).toBeCalled()
  })
})
