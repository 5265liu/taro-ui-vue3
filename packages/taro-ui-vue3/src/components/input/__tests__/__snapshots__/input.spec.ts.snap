// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtInput Snap render AtInput -- props border 1`] = `
<view
  class="at-input"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputFDJGLGEG"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props clear 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputIMJGKADG"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props disabled 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container at-input--disabled"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputDJMNNLBL"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props editable 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container at-input--disabled"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputBMFCMOPO"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props error 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container at-input--error"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputCENPAEGC"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <view
      class="at-input__icon"
    >
      <text
        class="at-icon at-icon-close-circle at-input__icon-alert"
      />
    </view>
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props maxLength 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputEDOEAFOE"
      maxlength="10"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props name 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputOIKGPDBE"
      maxlength="140"
      name="name"
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props placeholder 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputAAKIMHNK"
      maxlength="140"
      name=""
      password="false"
      placeholder="placeholder"
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props required 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputFDPNBJGP"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props style 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputCPBIGFNA"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props title 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <label
      class="at-input__title"
      for=""
    >
      title
    </label>
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputDPODGCOE"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props type(digit) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputNNODCCCM"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="digit"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props type(idcard) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputKNKKELND"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="idcard"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props type(number) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputFKGJBPKD"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="number"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props type(password) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputPIPDHGMK"
      maxlength="140"
      name=""
      password="true"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props type(phone) 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputKJMBEJJH"
      maxlength="11"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="number"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render AtInput -- props value 1`] = `
<view
  class="at-input at-input--without-border"
  value="value"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputABLPCDOH"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;

exports[`AtInput Snap render initial AtInput 1`] = `
<view
  class="at-input at-input--without-border"
>
  <view
    class="at-input__container"
  >
    <!-- overlay -->
    <view
      class="at-input__overlay at-input__overlay--hidden"
    />
    <!-- title -->
    <!--v-if-->
    <!-- input -->
    <input
      adjust-position="false"
      auto-focus="false"
      class="at-input__input"
      confirm-type="done"
      cursor="0"
      cursor-spacing="50"
      id="weui-inputAMCAGBLO"
      maxlength="140"
      name=""
      password="false"
      placeholder=""
      placeholder-class="[object Object]"
      placeholder-style=""
      selection-end="-1"
      selection-start="-1"
      type="text"
    />
    <!-- clear icon -->
    <!--v-if-->
    <!-- error icon -->
    <!--v-if-->
    <!-- input slot -->
    <view
      class="at-input__children"
    >
      
      
    </view>
  </view>
</view>
`;
